pkg/ledger/executor.go

```diff
@@ -123,6 +123,17 @@ func (l *Ledger) execute(ctx context.Context, script core.Script) (*core.Transac
 }
 
 func (l *Ledger) Execute(ctx context.Context, script core.Script) (*core.ExpandedTransaction, error) {
+
+	if script.Reference != "" {
+		txs, err := l.GetTransactions(ctx, *NewTransactionsQuery().WithReferenceFilter(script.Reference))
+		if err != nil {
+			return nil, err
+		}
+		if len(txs.Data) > 0 {
+			return nil, NewConflictError()
+		}
+	}
+
```

Comment:
why not add this check in the `execute` method above? it would avoid code duplication